test.py

```diff
@@ -587,28 +589,16 @@ async def main() -> int:
 
     setup_signal_handlers(asyncio.get_running_loop(), signaled)
 
-    tp_server = None
     try:
-        if [t for t in TestSuite.all_tests() if isinstance(t, LdapTest)]:
-            tp_server = subprocess.Popen('toxiproxy-server', stderr=subprocess.DEVNULL)
-            def can_connect_to_toxiproxy():
-                return can_connect(('127.0.0.1', 8474))
-            if not try_something_backoff(can_connect_to_toxiproxy):
-                raise Exception('Could not connect to toxiproxy')
-
-        try:
-            logging.info('running all tests')
-            await run_all_tests(signaled, options)
-            logging.info('after running all tests')
-            stop_event.set()
-            async with asyncio.timeout(5):
-                await resource_watcher
-        except Exception as e:
-            print(palette.fail(e))
-            raise
-    finally:
-        if tp_server is not None:
-            tp_server.terminate()
+        logging.info('running all tests')
+        await run_all_tests(signaled, options)
+        logging.info('after running all tests')
+        stop_event.set()
+        async with asyncio.timeout(5):
+            await resource_watcher
+    except Exception as e:
+        print(palette.fail(e))
+        raise
```

Comment:
The old code had here tp_server.terminate(). As far as I can tell, we don't have this in the new version. Is this ok?

Reply:
This is OK because the start and stop of toxiproxy moved completely to `start_3rd_party_services`